### ISSUE TITLE
Make SmvInputBase derive from a private class instead of SmvModule

### ISSUE THREAD

**Issue by ninjapapa** [TresAmigosSD/SMV#1465]
There are client code has the following logic:
```python
...
        elif isinstance(ds, smv.SmvModule):
            ......
        elif isinstance(ds, smv.SmvCsvFile):
            ......
```

Since we switched `SmvInputBase`'s super class from `SmvDataSet` to `SmvModule` we changed above logic's result. To keep the client code happy (although the client code itself should be fixed), we will rename `SmvModule` to `SmvSparkDfModule`, and make both `SmvModule` and `SmvInputBase` derive from it.

**Comment by ninjapapa** [comment]
Since PR #1461 hasn't been merged yet. Made the fix commit of this issue on the same PR.

**Comment by AliTajeldin** [comment]
Makes sense 👍

**Comment by ninjapapa** [comment]
Fixed in #1461
